common/src/main/java/com/gregtechceu/gtceu/common/machine/multiblock/electric/PowerSubstationMachine.java

```diff
@@ -0,0 +1,538 @@
+import com.gregtechceu.gtceu.api.machine.feature.IFancyUIMachine;
+import com.gregtechceu.gtceu.api.machine.feature.multiblock.IDisplayUIMachine;
+import com.gregtechceu.gtceu.api.machine.feature.multiblock.IMaintenanceMachine;
+import com.gregtechceu.gtceu.api.machine.feature.multiblock.IMultiPart;
+import com.gregtechceu.gtceu.api.machine.multiblock.IBatteryData;
+import com.gregtechceu.gtceu.api.machine.multiblock.WorkableMultiblockMachine;
+import com.gregtechceu.gtceu.api.machine.trait.MachineTrait;
+import com.gregtechceu.gtceu.api.machine.trait.RecipeLogic;
+import com.gregtechceu.gtceu.api.misc.EnergyContainerList;
+import com.gregtechceu.gtceu.config.ConfigHolder;
+import com.gregtechceu.gtceu.utils.FormattingUtil;
+import com.lowdragmc.lowdraglib.gui.modular.ModularUI;
+import com.lowdragmc.lowdraglib.gui.widget.*;
+import com.lowdragmc.lowdraglib.syncdata.field.ManagedFieldHolder;
+import it.unimi.dsi.fastutil.longs.Long2ObjectMaps;
+import lombok.Getter;
+import net.minecraft.ChatFormatting;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.HoverEvent;
+import net.minecraft.network.chat.Style;
+import net.minecraft.world.entity.player.Player;
+import org.jetbrains.annotations.NotNull;
+
+import java.math.BigInteger;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class PowerSubstationMachine extends WorkableMultiblockMachine implements IFancyUIMachine, IDisplayUIMachine {
+    protected static final ManagedFieldHolder MANAGED_FIELD_HOLDER = new ManagedFieldHolder(PowerSubstationMachine.class, WorkableMultiblockMachine.MANAGED_FIELD_HOLDER);
+
+    // Structure Constants
+    public static final int MAX_BATTERY_LAYERS = 18;
+    public static final int MIN_CASINGS = 14;
+
+    // Passive Drain Constants
+    // 1% capacity per 24 hours
+    public static final long PASSIVE_DRAIN_DIVISOR = 20 * 60 * 60 * 24 * 100;
+    // no more than 100kEU/t per storage block
+    public static final long PASSIVE_DRAIN_MAX_PER_STORAGE = 100_000L;
+
+    // Match Context Headers
+    public static final String PMC_BATTERY_HEADER = "PSSBattery_";
+
+    private static final BigInteger BIG_INTEGER_MAX_LONG = BigInteger.valueOf(Long.MAX_VALUE);
+
+    private IMaintenanceMachine maintenance;
+
+    private PowerStationEnergyBank energyBank;
+    private EnergyContainerList inputHatches;
+    private EnergyContainerList outputHatches;
+    private long passiveDrain;
+
+    // Stats tracked for UI display
+    private long netIOLastSec;
+    @Getter
+    private long averageIOLastSec;
+
+    protected ConditionalSubscriptionHandler tickSubscription;
+
+    public PowerSubstationMachine(IMachineBlockEntity holder) {
+        super(holder);
+        this.tickSubscription = new ConditionalSubscriptionHandler(this, this::transferEnergyTick, this::isFormed);
```

Comment:
This should probably use `isSubscriptionActive` instead, but I'm not 100% sure.

```suggestion
        this.tickSubscription = new ConditionalSubscriptionHandler(this, this::transferEnergyTick, this::isSubscriptionActive);
```

Though `isSubscriptionActive` would probably also need to check for `isFormed()` in that case.